doc/reference/trd-legal.md

```diff
@@ -0,0 +1,183 @@
+Licensing and Copyrights
+========================================
+
+**TRD:** <br/>
+**Working Group:** Core<br/>
+**Type:** Best Current Practice<br/>
+**Status:** Draft <br/>
+**Author:** Pat Pannuto<br/>
+**Draft-Created:** 2022/11/05 <br/>
+**Draft-Modified:** 2022/11/05 <br/>
+**Draft-Version:** 1 <br/>
+**Draft-Discuss:** tock-dev@googlegroups.com<br/>
+
+Abstract
+--------
+
+This document describes Tock’s policy on licensing and copyright. It explains
+the rationale behind the license selection (dual-license, MIT or Apache2) and
+copyright policy (optional, and authors may retain copyright if desired). It
+further outlines how licensing and copyright shall be handled throughout
+projects under the Tock umbrella.
+
+Explicitly not discussed in this TRD are issues of trademark, the Tock brand,
+logos, or other non-code assets and artifacts.
+
+
+1. Introduction
+===============
+
+Tock’s goal is to provide a safe, secure, and efficient operating system for
+microcontroller-class devices. The Tock project further believes it is
+important to enable and support widespread adoption of safe, secure, and
+efficient software. Tock also seeks to be an open and inclusive project, and
+Tock welcomes contributions from any individuals or entities wishing to
+improve the safety, security, reliability, efficiency, or usability of Tock
+and the Tock ecosystem.
+
+The intent of these policies is to best satisfy the needs of all stakeholders
+in the Tock ecosystem.
+
+
+2. Licensing
+============
+
+All software artifacts under the umbrella of the Tock project are
+dual-licensed as Apache License, Version 2.0 (LICENSE-APACHE or
+http://www.apache.org/licenses/LICENSE-2.0) or MIT license (LICENSE-MIT or
+http://opensource.org/licenses/MIT).
+
+All contributions to Tock projects MUST be licensed under these terms.
```

Comment:
This doesn't seem right.

What about tools? What if there are tools that run on the host. For example scripts that we might use to run tests, checks or flash binaries.

Also, what about newlib which is used by libtock-c? The binaries are [committed to a Tock project](https://github.com/tock/libtock-c/tree/master/newlib/cortex-m/v7-m)

Reply:
I do think it is a good question whether this should actually apply to all Tock project code -- for example, I don't think we want to go and add copyright headers to every file in the tock-teensy repo. Similarly, a tool like tockloader is a published artifact, and even most corporate users will not be depending on any of its source files directly -- they just might invoke the tool at some point during the process of building/flashing.

@jrvanwhy for Google's purposes, do you know what subset of Tock project repositories need to have license headers? For example, would just the kernel be sufficient? Or the kernel + libtock-c? I think we should limit the scope of this as much as possible.

Reply:
Conclusion 1: It's a lot easier if everything in a repository is under the same/compatible licenses. Let's keep it to that for Tock.

Conclusion 2: For other repositories, let's punt with this document. I.e. change the language to refer _specifically_ to _this_ Tock repository, not the Tock "project" as a whole.

Reply:
This PR has been updated to only apply to the `tock` repository. We expect to discuss applying this policy to other repositories after this PR has been merged.